lib/utils/io.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import "github.com/gravitational/trace"
+
+// WriteAll writes a buffer data to a writer until all data is written or an error is encountered.
+func WriteAll(w func(p []byte) (int, error), data []byte) error {
+	for len(data) > 0 {
```

Comment:
This loop will never execute more than one iteration, unless you give it a non-conforming `io.Writer`. This is because `Write` must return a non-nil error if it wrote less data than you asked it to, in which case you'll return an error.